analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -40,37 +41,32 @@ public class CheckArgumentException : SonarDiagnosticAnalyzer
         private const string ConstructorParametersInverted = "ArgumentException constructor arguments have been inverted.";
         private const string InvalidParameterName = "The parameter name '{0}' is not declared in the argument list.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         private static readonly ImmutableArray<KnownType> ArgumentExceptionTypesToCheck =
             ImmutableArray.Create(
                 KnownType.System_ArgumentException,
                 KnownType.System_ArgumentNullException,
                 KnownType.System_ArgumentOutOfRangeException,
-                KnownType.System_DuplicateWaitObjectException
-            );
+                KnownType.System_DuplicateWaitObjectException);
 
         protected override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterSyntaxNodeActionInNonGenerated(CheckForIssue,
-                SyntaxKind.ObjectCreationExpression);
+            context.RegisterSyntaxNodeActionInNonGenerated(CheckForIssue, SyntaxKind.ObjectCreationExpression);
 
         private static void CheckForIssue(SyntaxNodeAnalysisContext analysisContext)
         {
             var objectCreationSyntax = (ObjectCreationExpressionSyntax)analysisContext.Node;
-            var methodSymbol = analysisContext.SemanticModel.GetSymbolInfo(objectCreationSyntax).Symbol
-                as IMethodSymbol;
-            if (methodSymbol?.ContainingType == null ||
-                !methodSymbol.ContainingType.IsAny(ArgumentExceptionTypesToCheck))
+            var methodSymbol = analysisContext.SemanticModel.GetSymbolInfo(objectCreationSyntax).Symbol as IMethodSymbol;
+            if (methodSymbol?.ContainingType == null || !methodSymbol.ContainingType.IsAny(ArgumentExceptionTypesToCheck))
             {
                 return;
             }
 
             if (objectCreationSyntax.ArgumentList.Arguments.Count == 0)
```

Comment:
This check is much faster than retrieving the method symbol and verifying it. Could we move it on top?

Reply:
No, we need to report on the expected exceptions. So the symbol loading is needed first. Otherwise this would report on any `new AnyTypeWithNoArgs()`